unitTests.js

```diff
@@ -0,0 +1,46 @@
+describe('NotesCollectionView', function(){
+  context('when a list item is added to a note', function(){
+    var collection = $el = view = null;
+
+    //setup the view
+    beforeEach(function(){
+      collection = new NotesCollection();
+      collection.createCallCount = 0;
+      collection.create = function(){
+        this.createCallCount++;
+      };
+      $el = $('<div />');
+      view = new NotesCollectionView({
+        collection: collection,
+        el: $el
+      });
+      view.render();
+    });
+
+    //add an input & click button
+    beforeEach(function(){
+      view.$('#notesInput').val('List Item');
+      view.$('#tskBtn').click();
+    })
+
+    //teardown the view
+    afterEach(function(){
+      $el.remove();
+    });
+
+    it('resets input field to nothing', function(){
+      chai.expect(view.$('#notesInput').val()).to.equal('');
```

Comment:
This particular thing is a fairly awkward edge case, even though (paradoxically) the underlying concept of attached/detached DOM fragments is a fairly fundamental concept. Browsers ship with an unfortunate number of these sorts of things, and that number goes up exponentially as 3rd-party libraries get added into the mix. I guess the closest thing I have to suggestions would be to:

1. assume the worst. If something seems broken or a little bit off, it probably is. Browsers _love_ defying our intuition. When a bug like this turns up, the approach is to isolate it and try to validate assumptions (In this case I checked the contents of the test and the implementation, and even hard-coded a behavior into the implementation to prove to myself that the test was behaving correctly, before concluding that the issue was in the implementation)

2. read up on unfamiliar browser behaviors as they turn up. When I run across a new API or technique I try to take a few moments to skim the MDN article and/or library documentation for context. It's not always directly helpful, but given all the esoteric details of the frontend a little reading is better than taking anything for granted.

So, not a single article so much as a bit of a routine to for developing and debugging—hopefully useful nonetheless!